tests/scripts/thread-cert/Cert_5_2_05_AddressQuery.py

```diff
@@ -219,17 +264,174 @@ def test(self):
-            command.check_address_notification(msg, self.nodes[DUT_REED], self.nodes[LEADER])
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        LEADER_MLEID = pv.vars['LEADER_MLEID']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        REED = pv.vars['REED']
+        REED_MLEID = pv.vars['REED_MLEID']
+        REED_RLOC = pv.vars['REED_RLOC']
+        MED = pv.vars['MED']
+        MED_MLEID = pv.vars['MED_MLEID']
+        MM = pv.vars['MM_PORT']
+        REED2001 = ''
+        REED2002 = ''
+        MED2001 = ''
+        MED2002 = ''
+
+        for addr in pv.vars['REED_IPADDRS']:
+            if addr.startswith(Bytes('2001')):
+                REED2001 = addr
+            if addr.startswith(Bytes('2002')):
+                REED2002 = addr
+
+        for addr in pv.vars['MED_IPADDRS']:
+            if addr.startswith(Bytes('2001')):
+                MED2001 = addr
+            if addr.startswith(Bytes('2002')):
+                MED2002 = addr
+
+        # Step 3: Verify topology is formed correctly except REED.
+
+        pv.verify_attached('ROUTER_2')
+        for i in range(1, 16):
+            with pkts.save_index():
+                if i == 2:
+                    continue
+                pv.verify_attached('ROUTER_%d' % i)
+
+        # Step 4: REED attaches to Router_1 and MUST NOT attempt to become
+        #         an active router by sending an Address Solicit Request
+
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(REED).\
+            filter_mle_cmd(MLE_PARENT_RESPONSE).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              CONNECTIVITY_TLV,
+                              LEADER_DATA_TLV,
+                              LINK_LAYER_FRAME_COUNTER_TLV,
+                              LINK_MARGIN_TLV,
+                              RESPONSE_TLV,
+                              SOURCE_ADDRESS_TLV,
+                              VERSION_TLV
+                               } <= set(p.mle.tlv.type)
+                   ).\
+                   must_next()
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(REED).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            filter(lambda p: {
+                              ADDRESS16_TLV,
+                              LEADER_DATA_TLV,
+                              NETWORK_DATA_TLV,
+                              SOURCE_ADDRESS_TLV,
+                              ROUTE64_TLV
+                              } <= set(p.mle.tlv.type) or\
+                             {
+                              ADDRESS16_TLV,
+                              LEADER_DATA_TLV,
+                              NETWORK_DATA_TLV,
+                              SOURCE_ADDRESS_TLV
+                              } <= set(p.mle.tlv.type)
+                   ).\
+                   must_next()
```

Comment:
I think it should be common to verify a device attaches to a specific device, maybe in a separate PR, add a common API to such such attachment verification?

Reply:
For some cases which purpose are not checking the nodes attachment, those tlv checks could be omitted since there are plenty of 5.1.x cases already verified these steps, thoughts?

Reply:
use verify_attached func for REED attaching ROUTER_1